### PR TITLE
Report worker crashes to summary command

### PR DESCRIPTION
**Pull request by ChrisBr**
Report worker crashes to summary command.

### CODE REVIEW

File: ruby/lib/ci/queue/redis/build_record.rb [hunk 1]
```diff
@@ -17,6 +17,17 @@ def queue_exhausted?
           @queue.exhausted?
         end
 
+        def report_worker_error(error)
+          redis.pipelined do |pipeline|
+            pipeline.hset(key('worker-errors'), config.worker_id, error.message)
```
Comment (casperisfine) [inline ruby/lib/ci/queue/redis/build_record.rb:22]: Should we clear this on retry?

Comment (ChrisBr) [reply ruby/lib/ci/queue/redis/build_record.rb:22]: Yeah good point.

Comment (ChrisBr) [reply ruby/lib/ci/queue/redis/build_record.rb:22]: Updated

File: ruby/lib/minitest/queue.rb [hunk 2]
```diff
@@ -268,8 +268,10 @@ def run_from_queue(reporter, *)
       exit!(41)
     rescue => error
       reopen_previous_step
+      queue.report_worker_error(error)
       puts red("This worker exited because of an uncaught application error:")
       puts red("#{error.class}: #{error.message}")
+      puts red("#{error.backtrace}")
```
Comment (casperisfine) [inline ruby/lib/minitest/queue.rb:274]: That will call `Array#to_s`, that's not what you want:

```
>> puts "#{["foo", "bar"]}"
["foo", "bar"]
```

```suggestion
      error.backtrace.each do |frame|
        puts red(frame)
      end
```

**Review by casperisfine — APPROVED** [review]

**Pull request merged**